ITE/6/README.md

```diff
@@ -0,0 +1,574 @@
+  "details": {
+    "timestamp": "2020-04-12T18:04:10Z",
+    "decision": "allow",
+    "environment": {
+      "gcp_project": "example-project",
+      "cluster": "us-east1-a.prod-cluster"
+    }
+  }
+}
+```
+
+# Motivation
+
+This ITE has two main goals:
+
+1.  Support [use cases](#motivating-use-case) where the existing link schema is
+    a poor fit. For example, test steps and vulnerability scans are not about
+    "producing" a new artifact so they are awkward to represent in the current
+    format.
+2.  Support interoperability with
+    [Binary Authorization](https://cloud.google.com/binary-authorization), which
+    will support the agreed-upon format once finalized. This way we have a
+    single ecosystem of software supply chain security.
+
+Functional requirements:
+
+*   Must support user-defined types and schemas, for two reasons:
+    *   To allow in-toto users to more naturally express attestations, as
+        explained above.
+    *   Because Binary Authorization does not want to require its users to use
+        the existing in-toto link schema, which is overly specific.
+*   Should allow indexing of attestations by artifact ID, without having to
+    understand the user-defined schema.
+    *   Primary reason: To support generic attestation indexing/storage/fetching
+        without requiring user configuration for every type of attestation.
+    *   Secondary reason: To simplify the programming model of policies. The
+        binding between artifact and attestation can be done in a framework
+        without requiring type-dependent configuration.
+    *   Implication: the association between attestation and primary artifact ID
+        must be standardized.
+*   Should allow identification of related artifacts IDs given an attestation,
+    without having to understand the user-defined schema.
+    *   Reason: To support "inline attestations," where the client fetches and
+        sends all required attestations to the server for verification. The
+        client does not know the policy ahead of time or understand all
+        attestation types.
+    *   Example: Given a provenance attestation for a docker image, it should be
+        possible to identify all the materials generically.
+    *   Implication: the association between attestation and related artifact
+        IDs must be standardized.
+
+Nonfunctional requirements:
+
+*   Must support backwards compatible links that can be consumed by existing
+    layout files.
+*   Must differentiate between different types of related artifacts (only if
+    related artifacts are standardized.) Examples: materials vs products,
+    sources vs build tools.
+    *   Should be type-dependent, rather than mandating "materials" and
+        "products."
+
+# Reasoning
+
+## Reason for `subject` and `materials` to be standard Attestation fields
+
+There are two main reasons for standardizing `subject` and `materials` within
+Attestation schema.
```

Comment:
Did I miss the fact that these are in fact standardized? Should this be mentioned earlier on?

Reply:
Clarified. PTAL.